tests/tools.vitruv.applications.pcmjava.tests/src/tools/vitruv/applications/pcmjava/tests/pcm2java/repository/OperationRequiredRoleMappingTransformationTest.xtend

```diff
@@ -0,0 +1,287 @@
+			)
+			.addImportWithNamespace(interfaceCompilationUnit)
+			.addPrivateField(getRequiredInterfacFieldOrVariableName(Pcm2JavaTestUtils.INTERFACE_NAME, Pcm2JavaTestUtils.COMPOSITE_COMPONENT_NAME), getReference(interfaceCompilationUnit))
+			.addConstructorInitalizationForField(getRequiredInterfacFieldOrVariableName(Pcm2JavaTestUtils.INTERFACE_NAME, Pcm2JavaTestUtils.COMPOSITE_COMPONENT_NAME))
+			.build
+			
+			assertExistenceOfCompilationUnitsDeeplyEqualTo(List.of(interfaceCompilationUnit, componentCompilationUnit))
+		]
+	}
+	
+	@Test
+	def void testChangeOperationRequiredRole() {
+		createRepository(Pcm2JavaTestUtils.REPOSITORY_NAME)
+		changePcmView [
+			modifySingleRepository [
+				val signature = createOperationSignature(Pcm2JavaTestUtils.OPERATION_SIGNATURE_1_NAME)
+				val interface1 = createOperationInterface(Pcm2JavaTestUtils.INTERFACE_NAME)
+				interface1.signatures__OperationInterface += signature
+				interfaces__Repository += interface1
+				val interface2 = createOperationInterface(Pcm2JavaTestUtils.INTERFACE_NAME + Pcm2JavaTestUtils.RENAME)
+				interfaces__Repository += interface2
+				val component1 = createBasicComponent(Pcm2JavaTestUtils.BASIC_COMPONENT_NAME)
+				component1.requiredRoles_InterfaceRequiringEntity += createOperationRequiredRole(interface1, component1)
+				components__Repository += component1
+				components__Repository += createBasicComponent(Pcm2JavaTestUtils.BASIC_COMPONENT_NAME + Pcm2JavaTestUtils.RENAME)
+			]
+		]
+		
+		changePcmView [
+			modifySingleRepository [
+				val component2 = claimComponent(Pcm2JavaTestUtils.BASIC_COMPONENT_NAME + Pcm2JavaTestUtils.RENAME)
+				val interface2 = claimOperationInterface(Pcm2JavaTestUtils.INTERFACE_NAME + Pcm2JavaTestUtils.RENAME)
+				val operationRequiredRole = claimComponent(Pcm2JavaTestUtils.BASIC_COMPONENT_NAME).requiredRoles_InterfaceRequiringEntity.filter(OperationRequiredRole).claimOne
+				operationRequiredRole.requiredInterface__OperationRequiredRole = interface2
+				component2.requiredRoles_InterfaceRequiringEntity += operationRequiredRole
+			]
+		]
+		
+		validateJavaView[
+			val interface1CompilationUnit = new FluentJavaInterfaceBuilder(Pcm2JavaTestUtils.INTERFACE_NAME, 
+				Pcm2JavaTestUtils.REPOSITORY_NAME + Pcm2JavaTestUtils.CONTRACTS_SUFIX
+				)
+				.addMethod(new MethodDescription(Pcm2JavaTestUtils.OPERATION_SIGNATURE_1_NAME, createVoid(), List.of()))
+				.build
+			val interface2CompilationUnit = new FluentJavaInterfaceBuilder(Pcm2JavaTestUtils.INTERFACE_NAME + Pcm2JavaTestUtils.RENAME, 
+				Pcm2JavaTestUtils.REPOSITORY_NAME + Pcm2JavaTestUtils.CONTRACTS_SUFIX
+				)
+				.build
+			val component1CompilationUnit = new FluentJavaClassBuilder(
+					Pcm2JavaTestUtils.BASIC_COMPONENT_NAME + Pcm2JavaTestUtils.IMPL_SUFIX,
+					Pcm2JavaTestUtils.REPOSITORY_NAME + "." + Pcm2JavaTestUtils.BASIC_COMPONENT_NAME
+				)
+				// it is probably also intended to remove an empty constructor (-> bug in reaction)
```

Comment:
Not an expert on the PCM-Java reactions but I would say it should be consistent, i.e. if for a newly created repository, there is a Java constructor, then there should be one after some attributes got removed. If not, it should be removed when deleting the last attribute.
Anyways, the placement of this comment is hard to detect. Either add a TODO comment in the reactions or create an appropriate issue please.